app/components/InternalLink/__tests__/index.test.js

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { MemoryRouter } from 'react-router-dom';
+import renderer from 'react-test-renderer';
+
+import InternalLink from '../index';
+
+function RouterWrappedLink(props) {
+  return (
+    <MemoryRouter>
+      <InternalLink {...props} />
+    </MemoryRouter>
+  );
+}
+
+describe('InternalLink component', () => {
+  test('matches the snapshot', () => {
+    const tree = renderer.create(<RouterWrappedLink to="/test" />).toJSON();
+
+    expect(tree).toMatchSnapshot();
+  });
+
+  test('matches the snapshot with existing state', () => {
+    const tree = renderer.create(<RouterWrappedLink to="/test-state" state={{ unrelated: 'stateval' }} />).toJSON();
+
+    expect(tree).toMatchSnapshot();
+  });
```

Comment:
None of these tests verify that the state has been modified correctly, resulting in the scroll reset flag being passed into the location state.

Reply:
Resolved in [f1739a6](https://github.com/chichiwang/tamsui/pull/119/commits/f1739a6245dfc1bea47b159ec8b83b3a79fc7a91)